R/git.R

```diff
@@ -479,11 +479,23 @@ git_sitrep <- function() {
   have_token <- have_github_token()
   if (have_token) {
     kv_line("Personal access token", "<found in env var>")
-    who <- github_user()
-    if (is.null(who)) {
-      ui_oops("Token is invalid.")
-      have_token <- FALSE
-    } else {
+
+    who <-
+      suppressMessages(
+        tryCatch(
+          gh::gh_whoami(),
```

Comment:
I think you should pass the github token explicitly here, which I believe will allow you to eliminate the `suppressMessages()`.

Reply:
And I'd suggest making this block

```R
who <- gh::gh_whoami()
kv_line("User", who$login)
kv_line("Name", who$name)
```

Then you can simplify the error handlers, and remove the if block below.

Reply:
Thanks for the suggestions - it looks much better now!